thali/thalireplicationmanager.js

```diff
@@ -179,27 +185,123 @@ function syncPeer(peer) {
         var options = { live: true, retry: true };
         this._replications[peer.peerIdentifier] = {
           from: this._db.replicate.from(remoteDB, options),
-          to: this._db.replicate.from(remoteDB, options)
-        }
+          to: this._db.replicate.to(remoteDB, options)
+        };
       }.bind(this));
     }
   }.bind(this));
-}
+};
+
+/**
+ * Retry a synchronization after a failture with the given peer identifier. This tears down
+ * all the client replications so that we can have a clean sync retry
+ * @param {String} peerIdentifier The peer identifier to retry the synchronization with.
+ */
+ThaliReplicationManager.prototype._syncRetry = function (peerIdentifier) {
+  if (this._isInRetry) { return; }
```

Comment:
Isn't this assuming that we can only be in retry with a single peer at a time? What if we have multiple peers we are simultaneously connecting to and we are in retry with one or more of them? Shouldn't isInRetry really be an array indexed by peerID?

Also while on the topic can we take clients and replications and the rest and collapse them into one array with an object? It would make it much easier to think about threading issues since all the places state can blow up are in one place?